Hide Scan Login Code when 2FA is active.

develric opened the pull request:
Fixes #17434

This PR has a companion FluxC PR [here](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2591) that adds the persistence of the 2FA flag in FluxC db.

That added flag is used in the MeFragment to hide/show the QR login entry.

NOTE: since we do not change that settings in the apps, we do not have an handy entry point where to trigger an update of the cache. Mitigated as reported in [this comment](https://github.com/wordpress-mobile/WordPress-Android/pull/17662#issuecomment-1351624407)

<details>
  <summary>This collapsable documents previous attempts considered for mitigation</summary>
~~I tried to mitigate this by chaining in the WPMainActivity a newFetchSettingsAction in the OnAccountChange event triggered (among others) by the OnResume -> newFetchAccountAction~~

~~The chaining in the WPMainActivity of a newFetchSettingsAction, was making connected tests for signup with magic link (e2eSignUpWithMagicLink) fail (at least most of the time). I'm not sure if it's a test glitch or there can be something more, but I decided to play safer and moved the newFetchSettingsAction mitigation to the onActivityResult in WPMainActivity for when the user navigates back from the Me screen. This seems good enough for the mention mitigation, but let me know wdyt 🙇~~
</details>





| QR login present | QR login NOT present |
|---|---|
| ![image](https://user-images.githubusercontent.com/47797566/206884996-4652486c-c653-4fad-b303-5c04162d0369.png) | ![image](https://user-images.githubusercontent.com/47797566/206885007-ed6ce5f1-c523-4765-88d3-6160869aec55.png) |

To test:

Testing the visibility of the QR login entry

- Using a .com test account, be sure the 2fa is not setup (or disable it for the test sake)
- Now install both the WP and the JP app
- Login in both with the same account and go in Me screen.
- Check the WP app is not showing the QR login entry whereas the JP app is showing the entry. You should be able to login via the QR code on the web (you can use another browser or an incognito window to avoid logging out from the web)
- Now enable the 2fa 
- In the JP app, go to the main My Site page and then in the Me screen again
- Check the QR login entry is not present

DB upgrade

- Remove the JP app
- Install the JP app from trunk and login. With AS App Inspection note the AccountModel table does not contain the 2fa field
- Now upgrade the JP app with code from this PR
- Open again the JP app and check with AS App Inspection that the 2fa field is now present in the AccountModel table

Check Content copy flow still works

- Install the 21.3 WP app (the 2fa field is not present in the AccountModel table), and login
- Remove the JP app if installed and install again from this PR
- Open the JP app; the Content copy flow should start
- Go through the flow and check it completes correctly
- With AS App Inspection check the 2fa field is present in the AccountModel table of the JP app and the content copy flow had no issue.
- Also check the visibility state of the qr code login menu is consistent with 2fa state

## Regression Notes
1. Potential unintended areas of impact
QR login entry not visible when expected.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
Updated existing unit test in FluxC.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

develric committed:
Adding a check if 2fa is active to show/hide the QR login in the JP app.

develric committed:
Adding a newFetchSettingsAction to get account settings for OnResume in WPMainActivity, so to get fresh data in case something on the settings (including 2fa settings) changed.

develric committed:
Link to fluxc commit.

develric committed:
Updating release notes.

develric committed:
Merge branch 'trunk' into issue/17434-scan-to-login-and-2fa

develric committed:
Fixing detekt.

develric committed:
Moving newFetchSettingsAction to coming back from Me screen.

develric committed:
Fixing checkstyle

RenanLukas reviewed (approved):
Thanks for the changes, @develric . All scenarios are working as expected :shipit:

develric committed:
Moving the newFetchSettingsAction call to onResume.

develric committed:
Linking to fluxc commit.

develric committed:
Merge branch 'trunk' into issue/17434-scan-to-login-and-2fa

develric commented:
While making some testing for 21.3 beta and the content copy flow, noticed that the newFetchSettingsAction is not getting invoked in the AppInitializer since the token is not active at the point in time when that call happens. I've moved the call to newFetchSettingsAction in WPMainActivity::onResume, and for me everything works, also tried a few times to run the connected tests locally and all were green for me. 

Please note I've added a dedicated test step for this in `Check Content copy flow still works` section

Ready for another pass @RenanLukas 🙇

develric committed:
Fixing checkstyle.

RenanLukas reviewed (approved):
Thanks for the changes, @develric.

I've tested it again and all the scenarios work as expected :shipit:

develric committed:
Link to FluxC trunk.

develric commented:
Targeted back this pr to 21.4 since my understa ding is that the code freeze has not happened yet
